worblehat-domain/src/main/java/de/codecentric/psd/worblehat/domain/StandardBookService.java

```diff
@@ -61,14 +68,25 @@ public Book findBookByIsbn(String isbn) {
 
 
     @Override
-    public Book createBook(String title, String author, String edition, String isbn, int yearOfPublication) {
+    public Optional<Book> createBook(@Nonnull String title,
+                                     @Nonnull String author,
+                                     @Nonnull String edition,
+                                     @Nonnull String isbn,
+                                     int yearOfPublication) {
         Book book = new Book(title, author, edition, isbn, yearOfPublication);
-        return bookRepository.save(book);
+
+        Set<Book> booksByIsbn = bookRepository.findBooksByIsbn(isbn);

```

Comment:
I wonder whether JPA is able to do something like: `Optional<Book> findFirstBookByIsbn(String)` This would again simplify this method.

Reply:
Indeed there is!